cpp/deps.bzl

```diff
@@ -0,0 +1,79 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def _clp_ext_com_github_nlohmann_json():
+    commit = "fec56a1a16c6e1c1b1f4e116a20e79398282626c"
+    commit_sha256 = "8cbda3504fd1624fbce641d3f6b884c76e5afead1fa6d6abfcbea4b734dc634b"
+    http_archive(
+        name = "clp_ext_com_github_nlohmann_json",
+        sha256 = commit_sha256,
+        urls = ["https://github.com/nlohmann/json/archive/{}.zip".format(commit)],
+        strip_prefix = "json-{}".format(commit),
+        add_prefix = "json",
+        build_file_content = """
+cc_library(
+    name = "libjson",
+    srcs = ["json/single_include/nlohmann/json.hpp"],
+    hdrs = ["json/single_include/nlohmann/json.hpp"],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+        """,
+    )
+
+def com_github_y_scope_clp():
+    _clp_ext_com_github_nlohmann_json()
+
+    commit = "084efa35b7e9a63aecc5e327b97aea2a1cef83bc"
+    commit_sha256 = "3aea613f00b8ca2e07803c5774a2faf8d7a315d983093eb4ce23a14a73414f72"
+    http_archive(
+        name = "com_github_y_scope_clp",
+        sha256 = commit_sha256,
+        urls = ["https://github.com/y-scope/clp/archive/{}.zip".format(commit)],
+        strip_prefix = "clp-{}".format(commit),
+        add_prefix = "clp",
+        build_file_content = """
+cc_library(
+    name = "libclp_ffi_core",
+    srcs = [
+        "clp/components/core/src/BufferReader.cpp",
+        "clp/components/core/src/ReaderInterface.cpp",
+        "clp/components/core/src/string_utils.cpp",
+        "clp/components/core/src/ffi/encoding_methods.cpp",
+        "clp/components/core/src/ffi/ir_stream/encoding_methods.cpp",
+        "clp/components/core/src/ffi/ir_stream/decoding_methods.cpp",
+    ],
+    hdrs = [
+        "clp/components/core/src/BufferReader.hpp",
+        "clp/components/core/src/Defs.h",
+        "clp/components/core/src/ErrorCode.hpp",
+        "clp/components/core/src/ReaderInterface.hpp",
+        "clp/components/core/src/string_utils.hpp",
+        "clp/components/core/src/string_utils.inc",
+        "clp/components/core/src/TraceableException.hpp",
+        "clp/components/core/src/type_utils.hpp",
+        "clp/components/core/src/ffi/encoding_methods.hpp",
+        "clp/components/core/src/ffi/encoding_methods.inc",
+        "clp/components/core/src/ffi/ir_stream/byteswap.hpp",
+        "clp/components/core/src/ffi/ir_stream/encoding_methods.hpp",
+        "clp/components/core/src/ffi/ir_stream/decoding_methods.hpp",
+        "clp/components/core/src/ffi/ir_stream/decoding_methods.inc",
+        "clp/components/core/src/ffi/ir_stream/protocol_constants.hpp",
+    ],
+    includes = ["."],
+    copts = [
+        "-std=c++20",
```

Comment:
(I'll investigate further into this.) As discussed offline, errors were reported without flag `"-mmacosx-version-min=10.15"` when building on Apple M1 Max with macOS Sonoma 14.5 (23F79):
```
ERROR: /private/var/tmp/_bazel_jliao8/4798110f3bf3172c2c1168fdeb337bac/external/com_github_y_scope_clp/BUILD.bazel:3:11: Compiling clp/components/core/src/clp/ffi/ir_stream/encoding_methods.cpp failed: (Exit 1): cc_wrapper.sh failed: error executing CppCompile command (from target @@com_github_y_scope_clp//:libclp_core) external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics ... (remaining 54 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_github_y_scope_clp/clp/components/core/src/clp/ffi/ir_stream/encoding_methods.cpp:3:
external/com_github_nlohmann_json/json/single_include/nlohmann/json.hpp:4384:57: error: 'path' is unavailable: introduced in macOS 10.15
void from_json(const BasicJsonType& j, std::filesystem::path& p)
                                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:446:33: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_EXPORTED_FROM_ABI path {
                                ^
In file included from external/com_github_y_scope_clp/clp/components/core/src/clp/ffi/ir_stream/encoding_methods.cpp:3:
external/com_github_nlohmann_json/json/single_include/nlohmann/json.hpp:5007:55: error: 'path' is unavailable: introduced in macOS 10.15
void to_json(BasicJsonType& j, const std::filesystem::path& p)
                                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:446:33: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_EXPORTED_FROM_ABI path {
                                ^
In file included from external/com_github_y_scope_clp/clp/components/core/src/clp/ffi/ir_stream/encoding_methods.cpp:3:
external/com_github_nlohmann_json/json/single_include/nlohmann/json.hpp:5009:11: error: 'string' is unavailable: introduced in macOS 10.15
    j = p.string();
          ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:856:39: note: 'string' has been explicitly marked unavailable here
  _LIBCPP_HIDE_FROM_ABI _VSTD::string string() const { return __pn_; }
                                      ^
3 errors generated.
Target //<REDACTED> failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 7.766s, Critical Path: 2.24s
INFO: 11 processes: 4 internal, 7 darwin-sandbox.
ERROR: Build did NOT complete successfully
FAILED: 
//<REDACTED>  FAILED TO BUILD
```

Reply:
Thanks @davidlion for sharing some pointers offline. After looking these docs:
1. https://bazel.build/reference/be/c-cpp
2. https://bazel.build/reference/be/functions#select
3. https://bazel.build/reference/be/common-definitions#configurable-attributes

we came up with this config:
```
    copts = [
        "-std=c++20",
    ] + select({
        "@platforms//os:osx": [
            "-mmacosx-version-min=10.15",
        ],
        "//conditions:default": [],
    }),
```